### PR TITLE
migrate volume/csi/csi_plugin.go logs to structured logging

### PR DESCRIPTION
**Pull request by CKchen0726**
What type of PR is this?
/kind cleanup

What this PR does / why we need it:
part of kubernetes/enhancements#1602

Which issue(s) this PR fixes:
keps/sig-instrumentation/1602-structured-logging

Structured Logging migration instructions

Special notes for your reviewer:

Does this PR introduce a user-facing change?

```release-notes
migrate volume/csi/csi_plugin.go logs to structured logging
```

### CODE REVIEW

File: pkg/volume/csi/csi_plugin.go [hunk 1]
```diff
@@ -96,8 +96,7 @@ var PluginHandler = &RegistrationHandler{}
 // ValidatePlugin is called by kubelet's plugin watcher upon detection
 // of a new registration socket opened by CSI Driver registrar side car.
 func (h *RegistrationHandler) ValidatePlugin(pluginName string, endpoint string, versions []string) error {
-	klog.Infof(log("Trying to validate a new CSI Driver with name: %s endpoint: %s versions: %s",
-		pluginName, endpoint, strings.Join(versions, ",")))
+	klog.InfoS(log("Trying to validate a new CSI Driver"), "name", pluginName, "endpoint", endpoint, "versions", strings.Join(versions, ","))
```
Comment (yangjunmyfm192085) [inline pkg/volume/csi/csi_plugin.go:99]:
```suggestion
	klog.InfoS(log("Trying to validate a new CSI Driver"), "name", pluginName, "endpoint", endpoint, "versions", strings.Join(versions, ","))
```

Comment (yangjunmyfm192085) [inline pkg/volume/csi/csi_plugin.go:99]: do we need log format?

File: pkg/volume/csi/csi_plugin.go [hunk 2]
```diff
@@ -302,7 +301,7 @@ func initializeCSINode(host volume.VolumeHost) error {
 			// using CSI for all Migrated volume plugins. Then all the CSINode initialization
 			// code can be dropped from Kubelet.
 			// Kill the Kubelet process and allow it to restart to retry initialization
-			klog.Fatalf("Failed to initialize CSINode after retrying: %v", err)
+			klog.ErrorS(err, "Failed to initialize CSINode after retrying")
```
Comment (yangjunmyfm192085) [inline pkg/volume/csi/csi_plugin.go:304]: maybe need add exit?

File: pkg/volume/csi/csi_plugin.go [hunk 3]
```diff
@@ -347,12 +346,12 @@ func (p *csiPlugin) RequiresRemount(spec *volume.Spec) bool {
 	}
 	driverName, err := GetCSIDriverName(spec)
 	if err != nil {
-		klog.V(5).Info(log("Failed to mark %q as republish required, err: %v", spec.Name(), err))
+		klog.V(5).ErrorS(err, log("Failed to mark spec as republish required"), "name", spec.Name())
```
Comment (yangjunmyfm192085) [inline pkg/volume/csi/csi_plugin.go:349]: InfoS?

File: pkg/volume/csi/csi_plugin.go [hunk 4]
```diff
@@ -347,12 +346,12 @@ func (p *csiPlugin) RequiresRemount(spec *volume.Spec) bool {
...
 		return false
 	}
 	csiDriver, err := p.csiDriverLister.Get(driverName)
 	if err != nil {
-		klog.V(5).Info(log("Failed to mark %q as republish required, err: %v", spec.Name(), err))
+		klog.V(5).ErrorS(err, log("Failed to mark spec as republish required"), "name", spec.Name())
```
Comment (yangjunmyfm192085) [inline pkg/volume/csi/csi_plugin.go:354]: InfoS?

File: pkg/volume/csi/csi_plugin.go [hunk 5]
```diff
@@ -1021,7 +1020,7 @@ func waitForAPIServerForever(client clientset.Interface, nodeName types.NodeName
 			// API server contacted
 			return true, nil
 		}
-		klog.V(2).Infof("Failed to contact API server when waiting for CSINode publishing: %s", lastErr)
+		klog.V(2).ErrorS(lastErr, "Failed to contact API server when waiting for CSINode publishing")
```
Comment (yangjunmyfm192085) [inline pkg/volume/csi/csi_plugin.go:1023]: InfoS?

File: pkg/volume/csi/csi_plugin.go (hunk 1 above)
Comment (CKchen0726) [reply pkg/volume/csi/csi_plugin.go:99]: I think that's fine to keep the status quo. As structured logging evolves we may discover a more reasonable solution to systematically adding annotations.

File: pkg/volume/csi/csi_plugin.go (hunk 2 above)
Comment (CKchen0726) [reply pkg/volume/csi/csi_plugin.go:304]: I will fix it

File: pkg/volume/csi/csi_plugin.go [hunk 6]
```diff
@@ -186,29 +185,29 @@ func (p *csiPlugin) Init(host volume.VolumeHost) error {
 
 	csiClient := host.GetKubeClient()
 	if csiClient == nil {
-		klog.Warning(log("kubeclient not set, assuming standalone kubelet"))
+		klog.InfoS(log("kubeclient not set, assuming standalone kubelet"))
```
Comment (xing-yang) [inline pkg/volume/csi/csi_plugin.go:188]: Is there a "warning" for structured logging?

File: pkg/volume/csi/csi_plugin.go [hunk 7]
```diff
@@ -284,11 +284,11 @@ func initializeCSINode(host volume.VolumeHost) error {
...
 		}
 		err = wait.ExponentialBackoff(initBackoff, func() (bool, error) {
-			klog.V(4).Infof("Initializing migrated drivers on CSINode")
+			klog.V(4).InfoS("Initializing migrated drivers on CSINode")
 			err := nim.InitializeCSINodeWithAnnotation()
 			if err != nil {
-				kvh.SetKubeletError(fmt.Errorf("failed to initialize CSINode: %v", err))
-				klog.Errorf("Failed to initialize CSINode: %v", err)
+				kvh.SetKubeletError(fmt.Errorf("Failed to initialize CSINode: %v", err))
+				klog.ErrorS(err, "Failed to initialize CSINode")
```
Comment (xing-yang) [inline pkg/volume/csi/csi_plugin.go:291]: It is not clear to me what is the official guideline on whether the starting letter should be upper or lower case in a message of the structured logging yet.  I'd like to see an official guideline on that.  See discussions in the following PR:

https://github.com/kubernetes-csi/external-provisioner/pull/639#discussion_r648307076

File: pkg/volume/csi/csi_plugin.go (hunk 6 above)
Comment (CKchen0726) [reply pkg/volume/csi/csi_plugin.go:188]: There is no WarningS, so reduce the log level

**Pull request closed without merging**